src/Resolvers/DataValidationRulesResolver.php

```diff
@@ -240,9 +241,19 @@ protected function resolveOverwrittenRules(
             $path
         );
 
-        $overwrittenRules = app()->call([$class->name, 'rules'], ['context' => $validationContext]);
+        $manualRules = app()->call([$class->name, 'rules'], ['context' => $validationContext]);
 
-        foreach ($overwrittenRules as $key => $rules) {
+        if ($this->shouldMergeRules($class)) {
+            $manualRules = collect($manualRules)->map(
+                fn (string|array $rules) => is_array($rules) ? $rules : explode('|', $rules)
+            )->all();
```

Comment:
Why not use the `RuleDenormalizer` here?

Also: when we add an attribute not to validate the property it will still be validated we don't want that.

I think a better solution would be to implement the logic within the loop where addRules isn't called since it replaces them but rather rules are merged over there (or not if the property shouldn"t be validated).

Reply:
I've moved this to wher you have suggested, and added a `merge` function on the `DataRules`